inference-engine/thirdparty/clDNN/tests/test_cases/set_output_memory_gpu_test.cpp

```diff
@@ -0,0 +1,368 @@
+            /*b1f2*/0.2f, 0.2f,  -10.f, 5.2f,
+            /*b1f3*/4.f,  0.5f,  8.f,   8.2f
+    };
+    set_values(input, input_vec);
+
+    network network(engine, topology);
+
+    network.set_input_data("input", input);
+    network.set_output_memory("reorder", output_mem);
+    auto outputs = network.execute();
+
+    auto output = outputs.at("reorder").get_memory();
+    EXPECT_TRUE(engine.is_the_same_buffer(*output_mem, *output));
+
+    cldnn::mem_lock<float> output_ptr(output, get_test_stream());
+    cldnn::mem_lock<float> output_mem_ptr(output_mem, get_test_stream());
+
+    for (size_t i = 0; i < output_ptr.size(); ++i) {
+        EXPECT_TRUE(are_equal(output_mem_ptr[i], output_ptr[i])) << i;
+    }
+}
+
+TEST(set_output_memory_gpu, top_k2) {
+    static const int32_t x_size = 2, y_size = 2, feature_num = 4, batch_num = 2;
+    auto& engine = get_test_engine();
+    const int top_k = 2;
+    auto input = engine.allocate_memory({ data_types::f32, format::bfyx,{ batch_num, feature_num, x_size , y_size } });
+    auto top_k_input = engine.allocate_memory({ data_types::f32, format::bfyx,{ 1, 1, 1, 1 } });
+    auto second_output = engine.allocate_memory({ data_types::f32, format::bfyx, { top_k, feature_num, x_size , y_size } });
+    auto second_output_mem = engine.allocate_memory({ data_types::f32, format::bfyx, { top_k, feature_num, x_size , y_size } });
+    topology topology;
+    topology.add(input_layout("input", input->get_layout()));
+    topology.add(cldnn::data("const", top_k_input));
+    topology.add(mutable_data("second_output", second_output));
+    topology.add(arg_max_min("arg_max", { "input", "const", "second_output" }, arg_max_min::min, top_k, arg_max_min::batch));
+    topology.add(reorder("reorder", "arg_max", second_output->get_layout()));
+
+    std::vector<float> input_vec = {
+            //y0x0 y0x1 y1x0 y1x1
+            /*b0f0*/0.1f, -0.1f, 0.9f,  1.5f,
+            /*b0f1*/0.2f, 0.2f,  -10.f, 5.2f,
+            /*b0f2*/0.2f, 0.2f,  -10.f, 5.2f,
+            /*b0f3*/0.2f, 0.2f,  -10.f, 4.2f,
+
+            /*b1f0*/3.f,  0.5f,  7.f,   10.f,
+            /*b1f1*/4.f,  0.5f,  8.f,   8.2f,
+            /*b1f2*/0.2f, 0.2f,  -10.f, 5.2f,
+            /*b1f3*/4.f,  0.5f,  8.f,   8.2f
+    };
+    set_values(input, input_vec);
+
+    network network(engine, topology);
+
+    network.set_input_data("input", input);
+    network.set_output_memory("reorder", second_output_mem);
+    auto outputs = network.execute();
+
+    auto output = outputs.at("reorder").get_memory();
+    EXPECT_TRUE(engine.is_the_same_buffer(*second_output_mem, *output));
+
+    cldnn::mem_lock<float> output_ptr(output, get_test_stream());
+    cldnn::mem_lock<float> output_mem_ptr(second_output_mem, get_test_stream());
+
+    for (size_t i = 0; i < output_ptr.size(); ++i) {
+        EXPECT_TRUE(are_equal(output_mem_ptr[i], output_ptr[i])) << i;
+    }
+}
+
+TEST(set_output_memory_gpu, basic_opt) {
+    GTEST_SKIP();
```

Comment:
Why?

Reply:
Because currently optimizer always keeps output reorder. So, this test works only if condition in remove_redundant_reorders is modified.
I failed to invent such test topology which works with current optimizer

Reply:
Why do we need this test then? Can it be removed?